camayoc/tests/qcs/api/credentials/test_network_creds.py

```diff
@@ -0,0 +1,192 @@
+        2) Update the network credential deleting password and adding sshkey.
+        3) Confirm network credential has been updated.
+    :expectedresults: The network credential is updated.
+    """
+    cred = Credential(
+        cred_type='network',
+        client=shared_client,
+        password=uuid4())
+    cred.create()
+    # add the id to the list to destroy after the test is done
+    cleanup.append(cred)
+    assert_matches_server(cred)
+
+    sshkeyfile = Path(uuid4())
+    sshkeyfile.touch()
+
+    cred.ssh_keyfile = str(sshkeyfile.resolve())
+    cred.password = None
+    cred.update()
+    assert_matches_server(cred)
+
+
+def test_update_sshkey_to_password(
+        shared_client, cleanup, isolated_filesystem):
+    """Create a network credential using password and switch it to use sshkey.
+
+    :id: d24a54b5-3d8c-44e4-a0ae-61584a15b127
+    :description: Create a network credential with a sshkey, then update it
+        to use a password.
+    :steps:
+        1) Create a network credential with a username and sshkey.
+        2) Update the network credential deleting sshkey and updating
+           password.
+        3) Confirm network credential has been updated.
+    :expectedresults: The network credential is updated.
+    """
+    ssh_keyfile = Path(uuid4())
+    ssh_keyfile.touch()
+
+    cred = Credential(
+        cred_type='network',
+        client=shared_client,
+        ssh_keyfile=str(ssh_keyfile.resolve()),
+    )
+    cred.create()
+    # add the id to the list to destroy after the test is done
+    cleanup.append(cred)
+    assert_matches_server(cred)
+    cred.client.response_handler = api.echo_handler
+    cred.password = uuid4()
+    cred.ssh_keyfile = None
+    cred.update()
+    assert_matches_server(cred)
+
+
+def test_negative_update_to_invalid(
+        shared_client, cleanup, isolated_filesystem):
+    """Attempt to update valid credential with invalid data.
+
+    :id: c34ea917-ee36-4b93-8907-24a5f87bbed3
+    :description: Create valid network credentials, then attempt to update to
+        be invalid.
+    :steps:
+        1) Create valid credentials with passwords or sshkey.
+        2) Update the network credentials:
+            a) using both password and sshkey
+            b) missing both password and sshkey
+    :expectedresults: Error codes are returned and the network credentials are
+        not updated.
+    """
+    ssh_keyfile = Path(uuid4())
+    ssh_keyfile.touch()
+
+    cred = Credential(
+        cred_type='network',
+        client=shared_client,
+        ssh_keyfile=str(ssh_keyfile.resolve()),
+    )
+    cred.create()
+    # add the id to the list to destroy after the test is done
+    cleanup.append(cred)
+    assert_matches_server(cred)
+
+    cred.client = api.Client(api.echo_handler)
+
+    # Try to update with both sshkeyfile and password
+    cred.password = uuid4()
+    response = cred.update()
+    assert response.status_code == 400
```

Comment:
Is there an expected message on the response? If yes what about asserting for it?